pkg/operator/k8sutil/deployment.go

```diff
@@ -107,3 +107,42 @@ func DeleteDeployment(clientset kubernetes.Interface, namespace, name string) er
 	}
 	return deleteResourceAndWait(namespace, name, "deployment", deleteAction, getAction)
 }
+
+// WaitForDeploymentImage waits for all deployments with the given labels are running.
+// WARNING:This is currently only useful for testing!
+func WaitForDeploymentImage(clientset kubernetes.Interface, namespace, label, container string, initContainer bool, desiredImage string) error {
```

Comment:
Isn't this method still only used for testing? Why did it move back to this package?

Reply:
Oh right! There is an import cycle if it exists in `test`, because I want to use test's `New(numberOfNodes)` function to simulate a k8s cluster with nodes. I didn't see any good reason to re-implement the exact same logic when I could move this back to its original home in k8sutil.

Reply:
In some sense I think it does make sense to be a k8sutil even though it's non-ideal right now. It just happens that we currently only use it for testing, but it is theoretically useful outside of tests.